src/main/java/com/faforever/client/player/FriendJoinedGameNotifier.java

```diff
@@ -49,11 +53,13 @@ void postConstruct() {
   @Subscribe
   public void onFriendJoinedGame(FriendJoinedGameEvent event) {
     Player player = event.getPlayer();
-    Game game = player.getGame();
+    Game game = event.getGame();
 
     audioService.playFriendJoinsGameSound();
 
-    if (preferencesService.getPreferences().getNotification().isFriendJoinsGameToastEnabled()) {
+    if (preferencesService.getPreferences().getNotification().isFriendJoinsGameToastEnabled()
+        && !eventList.contains(event)) {
+      eventList.add(event);
```

Comment:
Since the events are never removed from the list, this creates a memory leak. The notifier keeps a reference to all events, the events have a reference to the players and games, those have further references, preventing all these objects from being garbage collected.

I'm not sure why this was added; since each event is only fired exactly once, the `contains()` check should always return false. What happens if we just remove `eventList`?

Reply:
You're right, should've thought further than that easy fix. 

I put it in because, no, it doesn't fire exactly once. In fact, it fires every time someone joins or leaves a game where you have a friend. It also fired each time any change was made to the game (gametitle change, swapped teams, added mods, etc.). If youre unlucky, you have two or more friends in there and it goes completely haywire.

This is how it looks when i befriended someone hosting a popular Gap-game:
![sanchopanda joined game](https://user-images.githubusercontent.com/33601179/36121031-5111a2f2-1045-11e8-99a3-c737c60cc7c9.jpg)


But I've looked over the code in PlayerServiceImpl and worked out a fix (which makes eventList unnecessary)